openshift_tools/ircbot/openshift_sre/openshift_sre.py

```diff
@@ -0,0 +1,351 @@
+GSHEET_URL_REGEX = r'https://docs\.google\.com/spreadsheets/d/(?P<key>[a-zA-Z0-9_-]*)/edit#gid=(?P<gid>[0-9]*)'
+SNOW_TICKET_REGEX = r'.*(?P<ticket>(TASK|REQ|RITM|INC|PRB|CHG)[0-9]{7}).*'
+MONITORED_CHANNELS = []
+SNOW_URL = 'https://url.corp.redhat.com/OpenShift-SRE-Service-Request-Form'
+SNOW_SEARCH = 'https://redhat.service-now.com/surl.do?n='
+SNOW_QUEUE = 'https://redhat.service-now.com/nav_to.do?uri=%2Fhome_splash.do%3Fsysparm_direct%3Dtrue'
+APAC = {
+    'name': 'APAC',
+    'column': 'O',
+    'utc_start': 22,
+    'utc_end': 6
+}
+EMEA = {
+    'name': 'EMEA',
+    'column': 'M',
+    'utc_start': 6,
+    'utc_end': 14
+}
+NASA = {
+    'name': 'NASA',
+    'column': 'K',
+    'utc_start': 14,
+    'utc–end': 22
+}
+ONCALL = {
+    'name': 'ONCALL',
+    'column': 'C'
+}
+# Ordered list of shifts, i.e., APAC is the first shift of a given day and NASA is the last shift
+SHIFTS = [APAC, EMEA, NASA]
+# Specify the time and date for when a new shift period should be used
+SHIFT_CHANGE = {
+    'weekday': 4,
+    'hour': 11,
+    'tz': timezone('US/Eastern')
+}
+
+
+class SreSection(StaticSection):
+    """Defines the config section for OpenShift SRE"""
+    google_service_account_file = FilenameAttribute('google_service_account_file',
+                                                    directory=False,
+                                                    default='service_account.json')
+    """File generated from a Google Cloud project (https://console.developers.google.com).
+
+    In a project, generate a service_account file from
+    APIs & services > Credentials > Create credentials > Service account key.
+    Also, be sure to enable Google Drive API.
+    """
+
+
+def configure(config):
+    """Adds SreSection to config"""
+    config.define_section('openshift_sre', SreSection)
+    config.openshift_sre.configure_setting(
+        'google_service_account_file',
+        'Enter the path to your service_account.json file.'
+    )
+
+
+def setup(bot):
+    """Sets up config with defaults in the case of missing SreSection"""
+    bot.config.define_section('openshift_sre', SreSection)
+
+
+def debug(bot, message):
+    """Provides debugging messages to bot owner if enabled."""
+    if DEBUG:
+        bot.say(message, bot.config.core.owner)
+
+
+def read_google_sheet(bot, gsheet_url, channel):
+    """Parses a google sheet URL."""
+    if gsheet_url:
+        match = re.search(GSHEET_URL_REGEX, gsheet_url)
+        groups = match.groupdict()
+        try:
+            gsheet_client = authorize(service_file=bot.config.openshift_sre.google_service_account_file)
+        except ValueError:
+            bot.say('The Service Account JSON file doesn\'t appear to exist. See '
+                    'https://cloud.google.com/storage/docs/authentication#service_accounts for creating a service '
+                    'account. Then move to ' + bot.config.openshift_sre.google_service_account_file, channel)
+            return False
```

Comment:
I don't think pylint likes this. What are you doing with a False return? `return None` is better I think.